### DISCUSSION TITLE
[WIP] changes for next release 3.11.1

### DISCUSSION THREAD

**Discussion by nlohmann** [nlohmann/json#3640] [Announcements]
This thread collects all changes made since the 3.11.0 (#3639) release.

### Summary

[Version 3.11.0](https://github.com/nlohmann/json/releases/tag/v3.11.0) moved the user-defined string literals (UDL) into namespace `nlohmann::literals::json_literals` (see [documentation](https://json.nlohmann.me/api/macros/json_use_global_udls/)). Unfortunately, this namespace was not imported to the global namespace by default which broke code. This release fixes this bug.

All changes are backward-compatible. See [release notes of version 3.11.0](https://github.com/nlohmann/json/releases/tag/v3.11.0) for more information on other features.

:moneybag: Note you can **support this project** via [GitHub sponsors](https://github.com/sponsors/nlohmann) or [PayPal](https://paypal.me/nlohmann).

### :bug: Bug fixes

- Set default value of [`JSON_USE_GLOBAL_UDLS`](https://json.nlohmann.me/api/macros/json_use_global_udls/) and the CMake options `JSON_GlobalUDLs` to `1` to import the user-defined string literals from the `nlohmann::literals::json_literals` namespace into the global namespace by default.
  - #3644
  - #3645
  - #3646

**Comment by falbrechtskirchinger** [comment]
> Set default value of JSON_GLOBAL_UDLS to 1 to import the nlohmann::literals::json_literals namespace to the global namespace by default.

The macro is named `JSON_USE_GLOBAL_UDLS` and the CMake options `JSON_GlobalUDLs` (following precedent).

Technically, it'd be more accurate to say: ... to import the user-defined string literals from the `nlohmann::literals::json_literals` namespace into the global namespace ...
Not too important of a distinction for users, but if we ever add another literal, it has to be added explicitly.
